src/cdomains/apron/apronDomain.apron.ml

```diff
@@ -15,6 +15,12 @@ module M = Messages
     - C API docs PDF (alternative mathematical descriptions): https://antoinemine.github.io/Apron/doc/api/c/apron.pdf
     - heterogeneous environments: https://link.springer.com/chapter/10.1007%2F978-3-030-17184-1_26 (Section 4.1) *)
 
+let widening_thresholds_apron = lazy (
+  (* Adding double value of all constants so that we can establish for single variables that they are <= const*)
+  let t = List.append (!Cilfacade.widening_thresholds) (List.map (fun x-> 2*x) (!Cilfacade.widening_thresholds)) in
```

Comment:
> but that's not what Apron seems to do: it has interval constraints as x <= c still.

Interesting: It was needed for this example: tests/regression/36-apron/89-mine14.c. Without the doubling, it required adding an additional constant variable of value `100` to work, the guess being that it would maintain `x <= 100` via this constant. After adding the doubling, it now works. :thinking:

Reply:
Could also be that the apron constraint output just simplifies `x + x <= 2c` to `x <= c` for output, while internally still having the former and also doing the widening with that doubled constant. There's no mention of that anywhere though.